\documentclass[twoside,11pt]{article}

% +
%  Name:
%     sun236.tex

%  Purpose:
%     SUN documentation for ORAC-DR spectroscopy (SUN/236)

%  Authors:
%     Paul Hirst (JAC)
%     Tim Jenness (JAC)
%     Brad Cavanagh (JAC)

%  Copyright:
%     Copyright (C) 2003-2005 Particle Physics and Astronomy
%     Research Council. All Rights Reserved.

%  History:
%     $Log$
%     Revision 1.19  2006/01/11 00:45:23  bradc
%     fix typo
%
%     Revision 1.18  2005/11/04 02:39:50  bradc
%     fix syntax error
%
%     Revision 1.17  2005/10/18 02:23:56  bradc
%     add details on science recipes
%
%     Revision 1.16  2005/10/18 01:00:46  bradc
%     fix micron symbol
%
%     Revision 1.15  2005/10/18 00:58:30  bradc
%     remove old text from document
%
%     Revision 1.14  2005/10/18 00:16:28  bradc
%     incomplete modifications to bring sun236 up to the same level of quality as sun232
%
%     Revision 1.13  2004/05/27 21:40:20  bradc
%     updates for v4.1
%
%     Revision 1.12  2003/06/12 23:19:54  bradc
%     updates for v4.0
%
%     Revision 1.17  2003/06/12 22:21:28  bradc
%     June, not May
%
%     Revision 1.16  2003/04/25 01:29:20  bradc
%     updates for 4.0 release
%
%     Revision 1.15  2002/09/16 04:30:21  timj
%     Add stardocversion
%
%     Revision 1.14  2002/09/16 03:21:54  timj
%     Fix minor tweaks
%
%     Revision 1.13  2002/09/16 03:13:33  timj
%     Fix spelling mistake
%
%     Revision 1.12  2002/09/16 03:09:48  timj
%     Brad was correct :-) \ORACDR\ is meant to be a latex command.
%
%     Revision 1.11  2002/09/14 01:49:19  phirst
%     fix Brads LaTeX
%
%     Revision 1.10  2002/05/28 21:12:22  bradc
%     Clarified use of calibration options
%
%     Revision 1.9  2001/12/14 02:58:17  timj
%     Wrong copyright
%
%     Revision 1.8  2001/12/14 02:53:06  timj
%     Minor updates for V3.0-3
%
%     Revision 1.7  2001/12/14 02:20:54  timj
%     even more missing typos
%
%     Revision 1.6  2001/12/14 02:18:17  timj
%     stardoccopyright as 2001
%
%     Revision 1.5  2001/12/14 02:16:53  timj
%     couple of typos plus increment doc number
%
%     Revision 1.4  2001/12/13 01:32:21  phirst
%     Added FAINT_POINT_SOURCE
%
%     Revision 1.3  2001/11/28 03:18:28  timj
%     Add xref to other oracdr docs
%
%     Revision 1.2  2001/11/28 01:58:23  timj
%     - Fix spelling mistakes
%     - Add logo
%     - Remove DESCRIPTION paragraph headings
%     - Tweak some sections to be real lists
%     - Some verbatim tables were not formatted verbatim
%

%  Revision:
%     $Id$

% -


% ? Specify used packages
\usepackage{graphicx}        %  Use this one for final production.
% \usepackage[draft]{graphicx} %  Use this one for drafting.
% ? End of specify used packages

\pagestyle{myheadings}

% -----------------------------------------------------------------------------
% ? Document identification
% Fixed part
\newcommand{\stardoccategory}  {Starlink User Note}
\newcommand{\stardocinitials}  {SUN}
\newcommand{\stardocsource}    {sun\stardocnumber}
\newcommand{\stardoccopyright} {Copyright \copyright\ 2005 Particle Physics and Astronomy Research Council}


% Variable part - replace [xxx] as appropriate.
\newcommand{\stardocnumber}    {236.5}
\newcommand{\stardocauthors}   {Paul Hirst \\ Brad Cavanagh \\
                                Joint Astronomy Centre, Hilo, Hawaii}
\newcommand{\stardocdate}      {October 2005}
\newcommand{\stardoctitle}     {ORAC-DR -- spectroscopy data reduction}
\newcommand{\stardocversion}   {5.0}
\newcommand{\stardocmanual}    {User Guide}
\newcommand{\stardocabstract}  {{\footnotesize ORAC-DR} is a
general-purpose automatic data-reduction pipeline environment.  This
document describes its use to reduce spectroscopy data collected at the
United Kingdom Infrared Telescope (UKIRT) with the CGS4, UIST and Michelle
instruments, at the Anglo-Australian Telescope (AAT) with the IRIS2
instrument, and from the Very Large Telescope with ISAAC. It outlines the
algorithms used and how to make minor modifications of them, and how
to correct for errors made at the telescope.}

% ? End of document identification
% -----------------------------------------------------------------------------

% +
%  Name:
%     sun.tex
%
%  Purpose:
%     Template for Starlink User Note (SUN) documents.
%     Refer to SUN/199
%
%  Authors:
%     AJC: A.J.Chipperfield (Starlink, RAL)
%     BLY: M.J.Bly (Starlink, RAL)
%     PWD: Peter W. Draper (Starlink, Durham University)
%
%  History:
%     17-JAN-1996 (AJC):
%        Original with hypertext macros, based on MDL plain originals.
%     16-JUN-1997 (BLY):
%        Adapted for LaTeX2e.
%        Added picture commands.
%     13-AUG-1998 (PWD):
%        Converted for use with LaTeX2HTML version 98.2 and
%        Star2HTML version 1.3.
%      1-FEB-2000 (AJC):
%        Add Copyright statement in LaTeX
%     {Add further history here}
%
% -

\newcommand{\stardocname}{\stardocinitials /\stardocnumber}
\markboth{\stardocname}{\stardocname}
\setlength{\textwidth}{160mm}
\setlength{\textheight}{230mm}
\setlength{\topmargin}{-2mm}
\setlength{\oddsidemargin}{0mm}
\setlength{\evensidemargin}{0mm}
\setlength{\parindent}{0mm}
\setlength{\parskip}{\medskipamount}
\setlength{\unitlength}{1mm}

% -----------------------------------------------------------------------------
%  Hypertext definitions.
%  ======================
%  These are used by the LaTeX2HTML translator in conjunction with star2html.

%  Comment.sty: version 2.0, 19 June 1992
%  Selectively in/exclude pieces of text.
%
%  Author
%    Victor Eijkhout                                      <eijkhout@cs.utk.edu>
%    Department of Computer Science
%    University Tennessee at Knoxville
%    104 Ayres Hall
%    Knoxville, TN 37996
%    USA

%  Do not remove the %begin{latexonly} and %end{latexonly} lines (used by
%  LaTeX2HTML to signify text it shouldn't process).
%begin{latexonly}
\makeatletter
\def\makeinnocent#1{\catcode`#1=12 }
\def\csarg#1#2{\expandafter#1\csname#2\endcsname}

\def\ThrowAwayComment#1{\begingroup
    \def\CurrentComment{#1}%
    \let\do\makeinnocent \dospecials
    \makeinnocent\^^L% and whatever other special cases
    \endlinechar`\^^M \catcode`\^^M=12 \xComment}
{\catcode`\^^M=12 \endlinechar=-1 %
 \gdef\xComment#1^^M{\def\test{#1}
      \csarg\ifx{PlainEnd\CurrentComment Test}\test
          \let\html@next\endgroup
      \else \csarg\ifx{LaLaEnd\CurrentComment Test}\test
            \edef\html@next{\endgroup\noexpand\end{\CurrentComment}}
      \else \let\html@next\xComment
      \fi \fi \html@next}
}
\makeatother

\def\includecomment
 #1{\expandafter\def\csname#1\endcsname{}%
    \expandafter\def\csname end#1\endcsname{}}
\def\excludecomment
 #1{\expandafter\def\csname#1\endcsname{\ThrowAwayComment{#1}}%
    {\escapechar=-1\relax
     \csarg\xdef{PlainEnd#1Test}{\string\\end#1}%
     \csarg\xdef{LaLaEnd#1Test}{\string\\end\string\{#1\string\}}%
    }}

%  Define environments that ignore their contents.
\excludecomment{comment}
\excludecomment{rawhtml}
\excludecomment{htmlonly}

%  Hypertext commands etc. This is a condensed version of the html.sty
%  file supplied with LaTeX2HTML by: Nikos Drakos <nikos@cbl.leeds.ac.uk> &
%  Jelle van Zeijl <jvzeijl@isou17.estec.esa.nl>. The LaTeX2HTML documentation
%  should be consulted about all commands (and the environments defined above)
%  except \xref and \xlabel which are Starlink specific.

\newcommand{\htmladdnormallinkfoot}[2]{#1\footnote{#2}}
\newcommand{\htmladdnormallink}[2]{#1}
\newcommand{\htmladdimg}[1]{}
\newcommand{\hyperref}[4]{#2\ref{#4}#3}
\newcommand{\htmlref}[2]{#1}
\newcommand{\htmlimage}[1]{}
\newcommand{\htmladdtonavigation}[1]{}

\newenvironment{latexonly}{}{}
\newcommand{\latex}[1]{#1}
\newcommand{\html}[1]{}
\newcommand{\latexhtml}[2]{#1}
\newcommand{\HTMLcode}[2][]{}

%  Starlink cross-references and labels.
\newcommand{\xref}[3]{#1}
\newcommand{\xlabel}[1]{}

%  LaTeX2HTML symbol.
\newcommand{\latextohtml}{\LaTeX2\texttt{HTML}}

%  Define command to re-centre underscore for Latex and leave as normal
%  for HTML (severe problems with \_ in tabbing environments and \_\_
%  generally otherwise).
\renewcommand{\_}{\texttt{\symbol{95}}}

% -----------------------------------------------------------------------------
%  Debugging.
%  =========
%  Remove % on the following to debug links in the HTML version using Latex.

% \newcommand{\hotlink}[2]{\fbox{\begin{tabular}[t]{@{}c@{}}#1\\\hline{\footnotesize #2}\end{tabular}}}
% \renewcommand{\htmladdnormallinkfoot}[2]{\hotlink{#1}{#2}}
% \renewcommand{\htmladdnormallink}[2]{\hotlink{#1}{#2}}
% \renewcommand{\hyperref}[4]{\hotlink{#1}{\S\ref{#4}}}
% \renewcommand{\htmlref}[2]{\hotlink{#1}{\S\ref{#2}}}
% \renewcommand{\xref}[3]{\hotlink{#1}{#2 -- #3}}
%end{latexonly}
% -----------------------------------------------------------------------------
% ? Document specific \newcommand or \newenvironment commands.

% degrees symbol
\newcommand{\dgs}{\hbox{$^\circ$}}
\begin{htmlonly}
\newcommand{\dgs}{{\rawhtml &deg;}}
\end{htmlonly}

% arcminute symbol
\newcommand{\arcm}{\hbox{$^\prime$}}
\begin{htmlonly}
\newcommand{\arcm}{{\rawhtml &acute;}}
\end{htmlonly}

% arcsec symbol
\newcommand{\arcsec}{\arcm\hskip -0.1em\arcm}
\begin{htmlonly}
\newcommand{\arcsec}{{\rawhtml &quot;}}
\end{htmlonly}

% decimal-degree symbol
\newcommand{\udeg}{\hskip-0.3em\dgs\hskip-0.08em}
\begin{htmlonly}
\newcommand{\udeg}{{\rawhtml &deg;}}
\end{htmlonly}

% decimal-arcsecond symbol
\newcommand{\uarcs}{\hskip-0.27em\arcsec\hskip-0.02em}
\begin{htmlonly}
\newcommand{\uarcs}{{\rawhtml &quot;}}
\end{htmlonly}

% centre an asterisk
\newcommand{\lsk}{\raisebox{-0.4ex}{\rm *}}

% Software
\newcommand{\ARD}{{\footnotesize ARD}}
\newcommand{\CCDPACK}{{\footnotesize CCDPACK}}
\newcommand{\CONVERT}{{\footnotesize CONVERT}}
\newcommand{\CURSA}{{\footnotesize CURSA}}
\newcommand{\GAIA}{{\footnotesize GAIA}}
\newcommand{\EXTRACTOR}{\mbox{\footnotesize EXTRACTOR}}
\newcommand{\FIGARO}{\mbox{\footnotesize FIGARO}}
\newcommand{\KAPPA}{{\footnotesize KAPPA}}
\newcommand{\ORACDR}{{\footnotesize ORAC-DR}}
\newcommand{\PHOTOM}{{\footnotesize PHOTOM}}
\newcommand{\PISA}{{\footnotesize PISA}}
\newcommand{\POLPACK}{{\footnotesize POLPACK}}
\newcommand{\FITSref}{\htmladdnormallink{FITS}{http://fits.gsfc.nasa.gov/}}

% Telescopes and instruments
\newcommand{\AAT}{\htmladdnormallink{AAT}{http://www.aao.gov.au/}}
\newcommand{\JCMT}{\htmladdnormallink{JCMT}{http://www.jach.hawaii.edu/JACpublic/JCMT/}}
\newcommand{\UKIRT}{\htmladdnormallink{UKIRT}{http://www.jach.hawaii.edu/JACpublic/UKIRT/}}
\newcommand{\VLT}{\htmladdnormallink{VLT}{http://www.eso.org/instruments/}}

\newcommand{\INGRID}{\htmladdnormallink{INGRID}{http://www.ing.iac.es/Astronomy/instruments/ingrid/}}
\newcommand{\IRCAM}{\htmladdnormallink{IRCAM}{http://www.jach.hawaii.edu/JACpublic/UKIRT/instruments/ircam/ircam3.html}}
\newcommand{\IRIS}{\htmladdnormallink{IRIS2}{http://www.aao.gov.au/iris2/}}
\newcommand{\ISAAC}{\htmladdnormallink{ISAAC}{http://www.eso.org/instruments/isaac/}}
\newcommand{\Michelle}{\htmladdnormallink{Michelle}{http://www.jach.hawaii.edu/JACpublic/UKIRT/instruments/michelle/michelle.html}}
\newcommand{\UIST}{\htmladdnormallink{UIST}{http://www.jach.hawaii.edu/JACpublic/UKIRT/instruments/uist/uist.html}}
\newcommand{\UFTI}{\htmladdnormallink{UFTI}{http://www.jach.hawaii.edu/JACpublic/UKIRT/instruments/ufti/ufti.html}}
\newcommand{\FP}{\htmladdnormallink{Fabry-Perot}{http://www.jach.hawaii.edu/JACpublic/UKIRT/instruments/ufti/ufti_fp.html}}
\newcommand{\CGS}{\htmladdnormallink{CGS4}{http://www.jach.hawaii.edu/JACpublic/UKIRT/instruments/cgs4/cgs4.html}}
\newcommand{\NIRI}{\htmladdnormallink{NIRI}{http://www.gemini.edu/sciops/instruments/niri/NIRIIndex.html}}

\newcommand{\ESO}{\htmladdnormallink{ESO}{http://www.eso.org}}

% +
%  Name:
%     SST.TEX

%  Purpose:
%     Define LaTeX commands for laying out Starlink routine descriptions.

%  Language:
%     LaTeX

%  Type of Module:
%     LaTeX data file.

%  Description:
%     This file defines LaTeX commands which allow routine documentation
%     produced by the SST application PROLAT to be processed by LaTeX and
%     by LaTeX2html. The contents of this file should be included in the
%     source prior to any statements that make of the sst commnds.

%  Notes:
%     The commands defined in the style file html.sty provided with LaTeX2html
%     are used. These should either be made available by using the appropriate
%     sun.tex (with hypertext extensions) or by putting the file html.sty
%     on your TEXINPUTS path (and including the name as part of the
%     documentstyle declaration).

%  Authors:
%     RFWS: R.F. Warren-Smith (STARLINK)
%     PDRAPER: P.W. Draper (Starlink - Durham University)

%  History:
%     10-SEP-1990 (RFWS):
%        Original version.
%     10-SEP-1990 (RFWS):
%        Added the implementation status section.
%     12-SEP-1990 (RFWS):
%        Added support for the usage section and adjusted various spacings.
%     8-DEC-1994 (PDRAPER):
%        Added support for simplified formatting using LaTeX2html.
%     {enter_further_changes_here}

%  Bugs:
%     {note_any_bugs_here}

% -

%  Define length variables.
\newlength{\sstbannerlength}
\newlength{\sstcaptionlength}
\newlength{\sstexampleslength}
\newlength{\sstexampleswidth}

%  Define a \tt font of the required size.
\latex{\newfont{\ssttt}{cmtt10 scaled 1095}}
\html{\newcommand{\ssttt}{\tt}}

%  Define a command to produce a routine header, including its name,
%  a purpose description and the rest of the routine's documentation.
\newcommand{\sstroutine}[3]{
   \goodbreak
   \markboth{{\stardocname}~ --- #1}{{\stardocname}~ --- #1}
   \rule{\textwidth}{0.5mm}
   \vspace{-7ex}
   \newline
   \settowidth{\sstbannerlength}{{\Large {\bf #1}}}
   \setlength{\sstcaptionlength}{\textwidth}
   \setlength{\sstexampleslength}{\textwidth}
   \addtolength{\sstbannerlength}{0.5em}
% Changed to -1.0 from -2.0 because there is only space for one banner.
   \addtolength{\sstcaptionlength}{-1.0\sstbannerlength}
% Changed to -2.5 from -5.0 because there is only space for one banner.
   \addtolength{\sstcaptionlength}{-2.5pt}
   \settowidth{\sstexampleswidth}{{\bf Examples:}}
   \addtolength{\sstexampleslength}{-\sstexampleswidth}
   \parbox[t]{\sstbannerlength}{\flushleft{\Large {\bf #1}}}
   \parbox[t]{\sstcaptionlength}{\center{\Large #2}}
%   \parbox[t]{\sstbannerlength}{\flushright{\Large {\bf #1}}}
   \begin{description}
      #3
   \end{description}
}

%  Format the description section.
\newcommand{\sstdescription}[1]{\item[Description:] #1}

%  Format the usage section.
\newcommand{\sstusage}[1]{\pagebreak[3] \item[Usage:] \mbox{} \\[1.3ex] {\ssttt #1}}

%  Format the invocation section.
\newcommand{\sstinvocation}[1]{\item[Invocation:]\hspace{0.4em}{\tt #1}}

%  Format the arguments section.
\newcommand{\sstarguments}[1]{
   \item[Arguments:] \mbox{} \\
   \vspace{-3.5ex}
   \begin{description}
      #1
   \end{description}
}

%  Format the returned value section (for a function).
\newcommand{\sstreturnedvalue}[1]{
   \item[Returned Value:] \mbox{} \\
   \vspace{-3.5ex}
   \begin{description}
      #1
   \end{description}
}

%  Format the parameters section (for a ORAC-DR recipe).
\newcommand{\sstparameters}[1]{
   \goodbreak
   \item[Configurable Steering Parameters:] \mbox{} \\
   \vspace{-3.5ex}
   \begin{description}
      #1
   \end{description}
}

%  Format the examples section.
\newcommand{\sstexamples}[1]{
   \goodbreak
   \item[Examples:] \mbox{} \\
   \vspace{-3.5ex}
   \begin{description}
      #1
   \end{description}
}

%  Define the format of a subsection in a normal section.
\newcommand{\sstsubsection}[1]{ \item[{#1}] \mbox{} \\}

%  Define the format of a subsection in the examples section.
\newcommand{\sstexamplesubsection}[2]{\sloppy \item[\parbox{\sstexampleslength}{\ssttt #1}] \mbox{} \\ #2 }

%  Format the notes section.
\newcommand{\sstnotes}[1]{\pagebreak[3] \item[Notes:] \mbox{} \\[1.3ex] #1}

%  Provide a general-purpose format for additional (DIY) sections.
\newcommand{\sstdiytopic}[2]{\goodbreak \item[{\hspace{-0.35em}#1\hspace{-0.35em}:}] \mbox{} \\[1.3ex] #2}

%  Format the implementation status section.
\newcommand{\sstimplementationstatus}[1]{
   \pagebreak[3] \item[{Implementation Status:}] \mbox{} \\[1.3ex] #1}

%  Format the bugs section.
\newcommand{\sstbugs}[1]{\item[Bugs:] #1}

%  Specify a variant of the itemize environment where the top separation
%  is reduced.  It is needed because a \vspace is ignored in the
%  \sstitemlist command.
\newenvironment{sstitemize}{%
  \vspace{-4.3ex}\begin{itemize}}{\end{itemize}}

%  Format a list of items while in paragraph mode.
\newcommand{\sstitemlist}[1]{
  \mbox{} \\
  \vspace{-3.5ex}
  \begin{sstitemize}
     #1
  \end{sstitemize}
}

%  Format a list of items while in paragraph mode, and where there
%  is a heading, thus the negative vertical space is not needed.
\newcommand{\ssthitemlist}[1]{
  \mbox{} \\
  \vspace{-3.5ex}
  \begin{itemize}
     #1
  \end{itemize}
}

%  Define the format of an item.
\newcommand{\sstitem}{\item}

%  Now define html equivalents of those already set. These are used by
%  latex2html and are defined in the html.sty files.
\begin{htmlonly}

%  sstroutine.
   \newcommand{\sstroutine}[3]{
      \subsection{#1\xlabel{#1}-\label{#1}#2}
      \begin{description}
         #3
      \end{description}
   }

%  sstdescription
   \newcommand{\sstdescription}[1]{\item[Description:]
      \begin{description}
         #1
      \end{description}
      \\
   }

%  sstusage
   \newcommand{\sstusage}[1]{\item[Usage:]
      \begin{description}
         {\ssttt #1}
      \end{description}
      \\
   }

%  sstinvocation
   \newcommand{\sstinvocation}[1]{\item[Invocation:]
      \begin{description}
         {\ssttt #1}
      \end{description}
      \\
   }

%  sstarguments
   \newcommand{\sstarguments}[1]{
      \item[Arguments:] \\
      \begin{description}
         #1
      \end{description}
      \\
   }

%  sstreturnedvalue
   \newcommand{\sstreturnedvalue}[1]{
      \item[Returned Value:] \\
      \begin{description}
         #1
      \end{description}
      \\
   }

%  sstparameters
   \newcommand{\sstparameters}[1]{
      \item[Configurable Steering Parameters:] \\
      \begin{description}
         #1
      \end{description}
      \\
   }

%  sstexamples
   \newcommand{\sstexamples}[1]{
      \item[Examples:] \\
      \begin{description}
         #1
      \end{description}
      \\
   }

%  sstsubsection
   \newcommand{\sstsubsection}[1]{\item[{#1}]}

%  sstexamplesubsection
   \newcommand{\sstexamplesubsection}[2]{\item[{\ssttt #1}] #2}

%  sstnotes
   \newcommand{\sstnotes}[1]{\item[Notes:]
      \begin{description}
         #1
      \end{description}
      \\
   }

%  sstdiytopic
   \newcommand{\sstdiytopic}[2]{\item[{#1}:]
      \begin{description}
         #2
      \end{description}
      \\
   }

%  sstimplementationstatus
   \newcommand{\sstimplementationstatus}[1]{\item[Implementation Status:]
      \begin{description}
         #1
      \end{description}
      \\
   }

%  sstitemlist
   \newcommand{\sstitemlist}[1]{
      \begin{itemize}
         #1
      \end{itemize}
      \\
   }

%  ssthitemlist
   \newcommand{\ssthitemlist}[1]{
      \begin{itemize}
         #1
      \end{itemize}
      \\
   }
\end{htmlonly}

%  End of "sst.tex" layout definitions.

% ? End of document specific commands
% -----------------------------------------------------------------------------
%  Title Page.
%  ===========
\renewcommand{\thepage}{\roman{page}}
\begin{document}
\setcounter{secnumdepth}{5}
\thispagestyle{empty}

%  Latex document header.
%  ======================
\begin{latexonly}
   CCLRC / \textsc{Rutherford Appleton Laboratory} \hfill \textbf{\stardocname}\\
   {\large Science and Technology Facilities Council}\\
   {\large Starlink Project\\}
   {\large \stardoccategory\ \stardocnumber}
   \begin{flushright}
   \stardocauthors\\
   \stardocdate
   \end{flushright}
   \vspace{-4mm}
   \rule{\textwidth}{0.5mm}
   \vspace{5mm}
   \begin{center}
   {\Huge\textbf{\stardoctitle \\ [2.5ex]}}
   {\LARGE\textbf{\stardocversion \\ [4ex]}}
   {\Huge\textbf{\stardocmanual}}
   \end{center}
   \vspace{5mm}

% ? Add picture here if required for the LaTeX version.
%   e.g. \includegraphics[scale=0.3]{filename.ps}
\begin{center}
\includegraphics[width=1.0in]{sun236_logo.eps}
\end{center}
% ? End of picture

% ? Heading for abstract if used.
   \vspace{10mm}
   \begin{center}
      {\Large\textbf{Abstract}}
   \end{center}
% ? End of heading for abstract.
\end{latexonly}

%  HTML documentation header.
%  ==========================
\begin{htmlonly}
   \xlabel{}
   \begin{rawhtml} <H1> \end{rawhtml}
      \stardoctitle\\
      \stardocversion\\
      \stardocmanual
   \begin{rawhtml} </H1> <HR> \end{rawhtml}

% ? Add picture here if required for the hypertext version.
%   e.g. \includegraphics[scale=0.7]{filename.ps}
\includegraphics[width=1.0in]{sun236_logo.eps}
% ? End of picture

   \begin{rawhtml} <P> <I> \end{rawhtml}
   \stardoccategory\ \stardocnumber \\
   \stardocauthors \\
   \stardocdate
   \begin{rawhtml} </I> </P> <H3> \end{rawhtml}
      \htmladdnormallink{CCLRC / Rutherford Appleton Laboratory}
                        {http://www.cclrc.ac.uk} \\
      \htmladdnormallink{Science and Technology Facilities Council}
                        {http://www.stfc.ac.uk} \\
   \begin{rawhtml} </H3> <H2> \end{rawhtml}
      \htmladdnormallink{Starlink Project}{http://www.starlink.ac.uk/}
   \begin{rawhtml} </H2> \end{rawhtml}
   \htmladdnormallink{\htmladdimg{source.gif} Retrieve hardcopy}
      {http://www.starlink.ac.uk/cgi-bin/hcserver?\stardocsource}\\

%  HTML document table of contents.
%  ================================
%  Add table of contents header and a navigation button to return to this
%  point in the document (this should always go before the abstract \section).
  \label{stardoccontents}
  \begin{rawhtml}
    <HR>
    <H2>Contents</H2>
  \end{rawhtml}
  \htmladdtonavigation{\htmlref{\htmladdimg{contents_motif.gif}}
        {stardoccontents}}

% ? New section for abstract if used.
  \section{\xlabel{abstract}Abstract}
% ? End of new section for abstract
\end{htmlonly}

% -----------------------------------------------------------------------------
% ? Document Abstract. (if used)
%  ==================
\stardocabstract
% ? End of document abstract

% -----------------------------------------------------------------------------
% ? Latex Copyright Statement
%  =========================
\begin{latexonly}
\newpage
\vspace*{\fill}
\stardoccopyright
\end{latexonly}
% ? End of Latex copyright statement

% -----------------------------------------------------------------------------
% ? Latex document Table of Contents (if used).
%  ===========================================
  \newpage
  \begin{latexonly}
    \setlength{\parskip}{0mm}
    \tableofcontents
    \setlength{\parskip}{\medskipamount}
    \markboth{\stardocname}{\stardocname}
  \end{latexonly}
% ? End of Latex document table of contents
% -----------------------------------------------------------------------------

\cleardoublepage
\renewcommand{\thepage}{\arabic{page}}
\setcounter{page}{1}

% ? Main text

\section{\xlabel{introduction}Introduction\label{introduction}}

\ORACDR\ is a data-reduction pipeline operating at \UKIRT, \JCMT, and
the \AAT. It is part of the \htmladdnormallink{ORAC system}{http://www.stsci.edu/stsci/meetings/adassVII/bridgera.html}.
The pipeline reduces and displays multi-frame observations soon after
they are read from the detector. This allows observers to assess the
quality and suitability of their data in near real time. Yet \ORACDR\
is capable of producing publication-quality results.

\ORACDR\ is suitable for `offline' data reduction at your home
institution as well. There are many reasons why you may wish to use
\ORACDR\ in this fashion. For instance, you may have come back from
UKIRT with only the raw observations; or there was an error in a
telescope sequence (formerly an `exec') mixing the groups of
observations; or some data were reduced with a basic algorithm for
speed at the telescope, and now you want to do a more careful job.
\ORACDR\ is capable of reducing data from instruments not running
the pipeline at their respective telescopes. Hence \ORACDR\ is
available on Starlink.

\xref{SUN/230}{sun230}{} presents an overview of \ORACDR, general
facilities like its display system, and it explains the differences
between a pipeline and a traditional reduction package. Briefly,
\ORACDR\ uses a few data headers to direct the data reduction.
Amongst these headers is the name of a {\em recipe}. A recipe is
a series of high-level instructions such as ``reduce, extract and
flux-calibrate a spectrum'' or ``divide by a flat'' that reduces
an {\em observation\/} comprising one or more data frames. The
implementation of each of these instructions is through a piece
of Perl code---called a primitive---which calls Starlink packages such as
\xref{\CCDPACK}{sun139}{} and \xref{\KAPPA}{sun95}{}, to actually
do the processing of the bulk data.

This document describes how to use \ORACDR\ software on Starlink
to reduce data from the UKIRT spectroscopy instruments: \CGS\,
\UIST, and \Michelle; the AAT spectrometer \IRIS; and the
\ISAAC\ instrument on the Very Large Telescope (\VLT). It outlines
the various algorithms used in the recipes and includes detailed
recipe documentation in the appendix. Besides the standard reduction
recipes, this manual describes how you can customise recipes to
suit your preferences and how to correct errors in the headers
of your data frames.

There are complementary documents: \xref{SUN/232}{sun232}{} describes
the \ORACDR\ for imaging from \UFTI, \UIST, \IRCAM, \Michelle, \IRIS,
\ISAAC\ and \INGRID; \xref{SUN246}{sun246}{} describes the \ORACDR\
for integral field spectroscopy from \UIST; and \xref{SUN/231}{sun231}{}
addresses the reduction of SCUBA data with \ORACDR.

Those wishing to write their own recipes from scratch, or wanting to
apply \ORACDR\ to new instruments should consult \xref{SUN/233}{sun233}{}

\section{\xlabel{using_the_pipeline}Using the pipeline\label{using_the_pipeline}}

\subsection{\xlabel{setting_up_orac-dr}Setting up \ORACDR\label{setting_up_orac-dr}}

Before you can run the pipeline you have to tell \ORACDR\ for which
instrument you wish to reduce data, the observation date, the directory
containing the raw data, and where you want the processed data to be
written. For the following two there are two options.

\begin{itemize}

\item The first needs your data to conform to the directory-naming
convention of the instrument at UKIRT. This will be the case if you
simply unpack the archive written by the {\bf uktape} utility.
In this case enter:

\begin{verbatim}
      % setenv ORAC_DATA_ROOT <root_data_directory>
      % oracdr_<instrument> <date>
\end{verbatim}
where {\tt $<$root\_data\_directory$>$} is the directory in which you
unpacked the data from the tape, {\tt $<$instrument$>$} is either
{\tt cgs4} or {\tt uist}, and {\tt$<$date$>$} is the UT date in the
format YYYYMMDD. Note that each \texttt{\%} represents the UNIX shell's
prompt, which you do not type. The commands must be entered in the
above order.

For example, the standard location for raw \CGS\ data is {\tt
raw/ufti/YYYYMMDD/}, and {\tt reduced/ufti/YYYYMMDD/} for the
corresponding reduced data. If your data are stored in
{\tt /home/users/abc/data/UKIRT/raw/cgs4/20031022/} you should enter
the following:

\begin{verbatim}
      % setenv ORAC_DATA_ROOT /home/users/abc/data/UKIRT/
      % oracdr_cgs4 20031022
\end{verbatim}
to enable the pipeline for \CGS\ data taken on 2003 October 22.

Data taken from the AAT is handled differently, as there is no unified
directory structure for either raw or reduced data directories. For
\IRIS, \INGRID\ or \ISAAC\ data the best option is specifying where
the raw and reduced data directories are, as shown below. \ISAAC\
users should see
\begin{htmlonly}
\htmlref{ISAAC preliminary conversion.}{isaac_preliminary_conversion}
\end{htmlonly}
\begin{latexonly}
Section~\ref{isaac_preliminary_conversion}
\end{latexonly}
for a necessary preliminary naming conversion step.

\item The second option is to separately define the raw and reduced
data directories. Type the following:

\begin{verbatim}
      % oracdr_<instrument> <date>
      % setenv ORAC_DATA_IN <raw_data_directory>
      % setenv ORAC_DATA_OUT <reduced_data_directory>
\end{verbatim}

The directories can either be given as full paths or as relative
paths to the current working directory. Here is an example for UIST
data using full paths:
\begin{verbatim}
      % oracdr_uist 20040414
      % setenv ORAC_DATA_IN /home/bradc/data/oracdr/asteroid/night1
      % setenv ORAC_DATA_OUT /home/scratch/bradc/reduced
\end{verbatim}

\end{itemize}
In the first case {\tt \$ORAC\_DATA\_IN} and {\tt \$ORAC\_DATA\_OUT}
are still defined, but in terms of the root directory. For instance,
re-using the earlier example with \CGS\ for UT date 2003 October 22,
{\tt \$ORAC\_DATA\_IN} points to {\tt \$ORAC\_DATA\_ROOT/raw/cgs4/20031022/}.

\ORACDR\ operates in {\tt \$ORAC\_DATA\_OUT}, irrespective of
what your current working directory is when you invoke it. Your
current directory remains unchanged.

It is highly recommended to work in directories on disks local to
the computer running the pipeline. Processing over NFS-mounted drives
can be many times slower and can degrade the performance seen by
other users. Running \ORACDR\ on a Linux computer over NFS-mounted
drives can also lead to corrupted files, crashing of the pipeline,
or computer lockups.

\subsection{\xlabel{raw_data_formats_and_conversions}Raw Data
Formats and Conversions\label{raw_data_formats_and_conversions}}

Raw data take the form of multiple NDFs within an
\xref{HDS container file}{sun92}{} for UKIRT data, or individual
\FITSref\ files for \AAT, \INGRID\ and \ISAAC\ data. For \UIST they
comprise of one NDF for the data array and dynamic headers, such
as the start time of the exposure, and another for static headers.
For the HDS containers, initial reduction steps operate on each of
the NDFs individually, only merging them to a simple NDF once the
interleaving step is complete.

The \Michelle\ HDS container also has NDFs for the individual chop
beams. However, these cannot be merged until the data variance
is calculated from the individual beams. Michelle reduced chopped
data become simple NDFs once the recipe takes the difference of the
two beams.

\ORACDR\ automatically converts AAT FITS files into single NDFs in
{\tt \$ORAC\_DATA\_OUT} which retain the original FITS headers.
For INGRID, \ORACDR\ converts a multi-extension FITS file into
a multi-NDF HDS container file following UKIRT conventions.

\subsubsection{\xlabel{isaac_preliminary_conversion}ISAAC Preliminary
Conversion\label{isaac_preliminary_conversion}}

Since \ORACDR\ as yet cannot cope with \ISAAC\ file naming, which
uses the UT epoch instead of a sequence number, there is a special
C-shell script which must be invoked once, normally before the first
\ORACDR\ initialisation. If you enter

\begin{verbatim}
      % isaac2oracdr
\end{verbatim}
in a directory containing ISAAC FITS files, the command converts them
into NDFs with names adhering to the UKIRT convention. The earliest
file has observation number 1, and the observation number increments
for each FITS file in time order. The script copes with file names in
either the raw or archive nomenclature. It copes with data from
more than one night in a given directory, assigning each night its
own sequence of observation numbers; and it uses a common UT date
for observations in a single night spanning midnight UT. You should
put all of the calibration and target files for a given night in
the same directory.

\subsection{\xlabel{running_the_pipeline}Running the
pipeline\label{running_the_pipeline}}

To run the pipeline, you use the {\bf oracdr} command. This has a
number of qualifiers described fully in \xref{SUN/230}{sun230}{oracdr}.
There is online help too; enter

\begin{verbatim}
      % oracdr -h
\end{verbatim}
for a list of the options.

Unlike using \ORACDR\ at \UKIRT, you are unlikely to need the looping
({\tt -loop} option) for offline processing, as all the data exist. Thus
the most important qualifiers are {\tt -list} and {\tt -from}, which
specify the frames to process; and the recipe name.

\begin{verbatim}
      % oracdr -from 42
\end{verbatim}
will process frames c20031022\_00042 until the end of the night's
data (assuming the earlier {\bf oracdr\_cgs4} command), running
the recipes given by each frame's header (RECIPE keyword). More likely
is that you provide a list of selected observations. The following
example

\begin{verbatim}
      % oracdr -list 41:49,51:59 POINT_SOURCE
\end{verbatim}
processes frames from 41 to 49 inclusive and 51 to 59 inclusive,
invoking the \htmlref{POINT\_SOURCE}{POINT\_SOURCE} recipe, and
overriding the RECIPE header.

\begin{verbatim}
      % oracdr -list 5,6,11,12
\end{verbatim}
would reduce the frame 5, 6, 11 and 12. This is most likely to
be applicable to pairs of flats and arcs.

There is a hazard with the {\tt -list} option. Take care to select
a complete set of frames associated with an observation. A common
error is to accidentally include an arc frame not part of the
group. Check the log on the raw data directory; it has file
extension {\tt .nightlog}. If you do not have a log, it is easy to
create one.
\label{night_log}

\begin{verbatim}
      % oracdr -from 1 -nodisplay NIGHT_LOG
\end{verbatim}
This will create a log in {\tt \$ORAC\_DATA\_IN} for the current
UT date. For CGS4, the log will be named {\tt $<$date$>$.nightlog}.
For multi-mode instruments such as Michelle, UIST, IRIS2, or ISAAC,
there may be two log files created, one called {\tt $<$date$>$\_im.nightlog}
and another called {\tt $<$date$>$\_sp.nightlog}, depending on the
observing mode. In general mode-agnostic observations such as array
tests are taken under imaging mode and will show up in the
{\tt \_im} log, whereas science and calibration observations will
show up in the {\tt \_sp} log.

\subsection{\xlabel{graphical_initialisation_and_operation}Graphical initialisation and
operation\label{graphical_initialisation_and_operation}}

You may prefer the \ORACDR\ graphical interface called
\xref{{\bf xoracdr}}{sun230}{xoracdr}.
\latexonly{( See SUN/230.)}  It allows you to configure
ORAC-DR: set the instrument, UT date, raw and reduced directories; and
to run the pipeline with the various options.  It permits monitoring
of the primitives during execution of a recipe. {\bf xoracdr} offers
access to other facilities like
\htmlref{display control}{display} and recipe editing.  The in-built
documentation does not pertain to the GUI itself but to general
\ORACDR\ information, however, {\bf xoracdr} is straightforward to use
and explore.  While {\bf xoracdr} has some rough edges, it is popular
with many users.  To try it, enter

\begin{verbatim}
      % xoracdr &
\end{verbatim}

Once the tool appears, you should select an instrument from the menu on
the left, a UT date in the top centre, and raw and reduced directories
to the lower right.   The {\tt From:} and {\tt To:} refer to the
observation numbers to process.  When you are ready to reduce data,
click on the {\tt Start ORAC-DR} button.

\subsection{\xlabel{display}Display\label{display}}

\ORACDR\ optionally lets you inspect the raw frames, and the processed
data as they are created.  There is a variety of graphical
methods available, including histograms and contour plots, if you
choose a \xref{\KAPPA}{sun95}{} GWM widget.  Most people prefer a
simple scaled image display with \xref{\GAIA}{sun214}{}.  This offers
facilities to inspect and analyse the data, and both pixel and sky
co-ordinates of the cursor position are presented.  The selection of
frame types to display, where they should appear, and how they are
scaled are configurable using a simple text file or a special GUI tool
{\bf oracdisp}.  See \xref{SUN/230}{sun230}{display_system} for details
and examples.

Processing offline, there is less need to see the data displayed in real time.
If you wish to accelerate the processing switch off the display option.

\begin{verbatim}
      % oracdr -nodisplay ...
\end{verbatim}

\subsection{\xlabel{calibration_information}Calibration Information\label{calibration_information}}

\ORACDR\ records calibration information, such as arc frames, flat
fields, and the read noise, within index files, one for each type of
calibration information.  When the pipeline needs a calibration frame
it searches the index file for the best matching entry subject to a
set of rules. Each recipe reports the calibrations it has used.  If no
suitable calibration exists, the pipeline exits with an error message
stating this fact.  For further details see
\xref{SUN/230}{sun230}{calibration_selection}.
\begin{htmlonly}
Here is an
\htmlref{example index file.}{index_files}
\end{htmlonly}
\begin{latexonly}
Section~\ref{index_files} has an example of an index file.
\end{latexonly}

You can also select a specific calibration using the {\tt -calib}
command-line option, provided the chosen calibration has an entry
in the appropriate index file.  See
\begin{latexonly}
the section on
\end{latexonly}
\xref{calibration options}{sun230}{calibration_options}
\begin{latexonly}
in SUN/230
\end{latexonly}
for details and examples.

\subsubsection{\xlabel{available_calib}Available calibration methods\label{available_calib}}

The following calibration methods are available for spectroscopy
recipes.

\begin{itemize}

\item bias --- Use the given bias frame.

\item flat --- Use the given flat frame.

\item mask --- Use the given bad-pixel mask.

\item readnoise --- Use the given value for the detector readnoise
in electrons.

\end{itemize}

\subsection{\xlabel{log_files}Log files\label{log_files}}

In addition to presenting the progressing data reduction to an
\ORACDR\ X-window, \ORACDR, by default, retains a copy of the
processing steps and errors in a log file.  These logs are important
if something has gone wrong, and you have exited the X-window.
Information from the applications software can be included if you run
the pipeline with the {\tt -verbose} command-line option.  Logs also
serve as a record of the data processing.  Yet the log files are often
overlooked because they are hidden.  The log file is called
{\tt\$ORAC\_DATA\_OUT/.oracdr\_$<$number$>$}, where {\tt$<$number$>$}
is the current process identification.  The {\tt -log f} option to the
{\bf oracdr} command enables log-file creation.

See \xref{SUN/230}{sun230}{windows_and_output} for details of the
logging options.

\section{\xlabel{features_of_the_primitives}Features of the
Primitives\label{features_of_the_primitives}}

Primitives are the Perl scripts which actually call the applications
to do most of the data processing. All of the spectroscopy recipes
are, in principle, independent of the instrument. however, some
recipes are inappropriate; for example, the \htmlref{LAMP\_FLAT}{LAMP\_FLAT}
recipe is intended for flat-field generation for \NIRI, but not
for other instruments.

Not all the following steps apply to all recipes. Consult the
\htmlref{reference section}{recipes} to see summaries for each
recipe. The steps are presented in normal order of appearance.

The main primitives pertinent to each step are listed in bracketed
italics, should you wish to tailor the recipes. These are found in
the {\tt\$ORAC\_DIR/spectroscopy} tree, unless they start with
{\em general/}. Note that some may be instrument-specific
variants, either given explicitly or with the {\em$<$instrument$>$}
token, which means substitute the instrument name in uppercase.

While the following listed primitives form the bulk of the primitive
library, there are many not listed here, mostly those for
\begin{htmlonly}
\htmlref{recipe initialisation}{hello_primitives} called
&<&recipe&>&_HELLO\_.
\end{htmlonly}
\begin{latexonly}
recipe initialisation called $<$recipe$>$\_HELLO\_ (see
Section~\ref{hello_primitives} for more information),
\end{latexonly}
and for
\begin{htmlonly}
\htmlref{recipe steering}{steering_primitive}, which
control when to perform certain operations, called
$<$recipe$>$\_STEER\_ or $<$recipe$>$\_CONFIG\_.
\end{htmlonly}
\begin{latexonly}
recipe steering which control when to perform certain operations,
called $<$recipe$>$\_STEER\_ or $<$recipe$>$\_CONFIG\_ (see
Section~\ref{steering_primitive}).
\end{latexonly}
The first of these is normally left unchanged unless there is a need
to add more steering parameters. Other primitives not mentioned here
are tied closely with single recipes, usually to create and file
calibrations.

\subsection{\xlabel{preparation_of_single_frames}Preparation of Single
Frames\label{preparation_of_single_frames}}

{\em[\_REDUCE\_SINGLE\_FRAME\_, $<$instrument$>$/\_INSTRUMENT\_HELLO\_,
\_SPECTROSCOPY\_HELLO\_]}

\subsubsection{\xlabel{manipulation_of_raw_data}Manipulation of Raw
Data\label{manipulation_of_raw_data}}

The first step copies the raw data into {\tt\$ORAC\_DATA\_OUT}. For
instruments whose raw data is in \FITSref\ format, this step converts
the raw data into NDF. For \xref{HDS}{sun92}{} container files, these
are copied over into new HDS container files. \newline {\em[\_MAKE\_RAW\_FILE\_]}

\subsubsection{\xlabel{preliminaries}Preliminaries\label{preliminaries}}

There are a few operations applied to all frames. First, history
recording is switched on. It is recommended to leave this enabled,
since it provides a record of the processing steps of your final
spectra. Otherwise the pipeline becomes something of a black box.
use the \KAPPA\ command \xref{{\bf hislist}}{sun95}{HISLIST} to
list the history records. \newline {\em[\_TURN\_ON\_HISTORY\_]}

The next step is to set the origin of the frame so that frame pixels
retain the detector pixel indices. It then becomes possible to use
a full-sized bad-pixel mask or a flat field on a subset of a
detector's pixel grid. \newline {\em[\_SET\_ORIGIN\_]}

For CGS4 data taken before 2000 August 13, the slit angle in the
SANGLE header referred to the slit's physical position in the
instrument, and after this date it referred to the slit's angle
on the sky. For data taken before this date, the header value
internal to ORAC-DR is set to coorespond to the on-sky angle.
\newline {\em[CGS4/\_FIX\_SANGLE\_HEADER\_]}

For UIST data taken before 2002 December 2 and Michelle, raw data
units are converted from ADU per second to the \UKIRT\ standard
of total ADU per exposure. \newline {\em[UIST/\_DATA\_UNITS\_TO\_ADU\_,
MICHELLE/\_DATA\_UNITS\_TO\_ADU\_]}

For Michelle there is a validation check of the waveform used, comparing
the waveform name given in the headers with other metadata, and
recipes issue a warning if there is an inconsistency. For lowQ and
MedN2 data taken before 2004 March 09, the gratings were installed
the wrong way round in the cryostat, so these data are flipped
along the dispersion axis at this stage in the data reduction.
\newline {\em[MICHELLE/\_CHECK\_WAVEFORM\_,
MICHELLE/\_FLIP\_FLIPPED\_GRATING\_FRAMES\_]}

A \htmlref{night log}{night_log} is created or appended in {\tt\$ORAC\_DATA\_OUT}
for each frame processed.  This tabulates the main parameters of the
observation having first corrected defective or undefined headers.
\newline {\em[\_NIGHT\_LOG\_, $<$instrument$>$/\_FIX\_EXTRA\_HEADERS\_
for MICHELLE, CGS4, and UIST]}

\subsubsection{\xlabel{bad_pixels}Bad pixels\label{bad_pixels}}

The recipes apply a predetermined bad-pixel mask with the aim of
removing the bulk of `hot' and `cold' pixels.  This flags
approximately 0.4\% of \UIST\ and
\ISAAC\ pixels, 0.1\% of \IRIS\ pixels, and 5\% of
{\Michelle}'s pixels.
\newline {\em[\_MASK\_BAD\_PIXELS\_]}

Some of the instruments (\UIST, \CGS) array tests are run, typically
at the start of each night. As a part of these array tests a new
bad-pixel-mask is generated on-the-fly, using the predetermined one
as a basis. For \UIST\ the new bad-pixel-mask is generated from
a long-exposure dark observation, typically 100s. Any pixel that
is 5-$\sigma$ higher than the 3-$\sigma$ clipped mean or 1000-$\sigma$
lower than the 3-$\sigma$ clipped mean is flagged as bad. For \CGS\ any
pixel higher than 1700 or lower than 15 for a dark whose exposure time
is longer than 80 seconds, or higher than 1500 and lower than -100 for
a dark whose exposure time is 80 seconds or shorter, is flagged as
bad.

\paragraph{\xlabel{create_bad-pixel_mask}Creating a bad-pixel
mask\label{create_bad-pixel_mask}}

The easiest way to create your own bad-pixel mask for use with the
calibration system, is to run the \htmlref{MAKE\_BPM}{MAKE\_BPM}
recipe on a long-exposure dark (at least 20 seconds integration).  It
is possible to change the symmetric $\sigma$-clipping bounds in the
recipe (see primitive {\tt\_MAKE\_BPM\_BY\_SIGMA\_THRESHOLDING\_}).
You can tailor this primitive if you want more control, say to have
asymmetric rejection or more sophisticated definitions.
\newline {\em[\_MAKE\_BPM\_BY\_SIGMA\_THRESHOLDING\_]}

For better results, use the average of long dark frames taken across
two or three nights.  First, produce QUICK\_LOOK versions of the
long-exposure dark to flatten the NDF structure or convert the FITS
file.  Flag all pixels that are 5 standard deviations ($\sigma$) above
and below the 3-$\sigma$ clipped mean of the dark as ``bad'', then
multiply the resulting frame by zero so that the resulting bad-pixel
mask has data values of {\tt 0} and {\tt bad} only.  You can choose your
own thresholds.  Here is an example, using data from two nights of
CGS4 data and Starlink software.

\begin{verbatim}
      % oracdr_cgs4 20010101
      % setenv ORAC_DATA_OUT `pwd`
      % oracdr -list 4:4 QUICK_LOOK -nodisplay
      % oracdr_cgs4 20010102
      % setenv ORAC_DATA_OUT `pwd`
      % oracdr -list 4:4 QUICK_LOOK -nodisplay

      % kappa
      % add c20010101_00004_mraw c20010102_00004_mraw add_darks
      % cmult add_darks 0.5 av_darks
      % stats av_dark clip=3
      % thresh av_darks av_darks_thresh -49 58 bad bad
      % cmult av_darks_thresh 0 avbpm title=\"CGS4 bpm, January 2001\"
\end{verbatim}

In the above example the 3-$\sigma$ clipped mean was 4.27 and the standard deviation
was 10.727, resulting in $-$49 and 58 as the lower and upper thresholds.

Then you specify the bad-pixel mask on the command line.
\begin{verbatim}
      % oracdr -calib mask=avbpm ...
\end{verbatim}

\UIST\ has its own slightly different formula; see \htmlref{DARK\_AND\_BPM}{DARK\_AND\_BPM}
for details

{\em[UIST/\_FIND\_BAD\_PIXELS\_, UIST/\_FILE\_BAD\_PIXELS\_, \_FILE\_MASK\_]}

\subsubsection{\xlabel{readnoise_variance}Readnoise
Variance\label{readnoise_variance}}

After the bad pixel mask has been applied, the readnoise variance
is added into the VARIANCE component of the NDF. For all instruments,
the readnoise value is obtained from the calibration system, having
been previously calculted in the \htmlref{ARRAY\_TESTS}{ARRAY\_TESTS}
recipe. Since the readnoise value is stored in electrons, it must be
converted into analogue-to-digital units.

{\em[\_ADD\_READNOISE\_VARIANCE\_, \_CALCULATE\_NREADS\_NOISE\_FACTOR\_]}

\paragraph{CGS4 Readnoise}

For \CGS\ the readnoise from the calibration system is first
divided by a factor to take multiple non-destructive reads into account:

\[   RNE = RN / \frac{3.8 + 3.5 * e^{-1/8 * N_{ND}}}{3.8 + 3.5 * e^{-1/8}} \]

where $RNE$ is the readnoise per exposure, in electrons, $RN$ is the
readnoise in electrons, and $N_{ND}$ is the number of non-destructive reads.

The formula used to determine the variance due to readnoise is:

\[   V_{RN} = \frac{RNE^{2}}{N_e * gain^{2}} \]

where $RNE$ is the readnoise per exposure, in electrons, $N_e$ is the number of exposures
per integration, and $gain$ is the detector gain in electrons per ADU.

\paragraph{Michelle Readnoise}

For \Michelle\ the readnoise from the calibration system is
also first divided by a factor to take multiple non-destructive
reads into account:

\[   RNE = \frac{RN}{\sqrt{\frac{N_{ND} * ( N_{ND} + 1 )}{12 * ( N_{ND} + 1 )}}} \]

where $RNE$ is the readnoise per exposure, in electrons, $RN$ is the
readnoise in electrons, and $N_{ND}$ is the number of non-destructive reads.

The formula used to determine the variance due to readnoise is:

\[   V_{RN} = \frac{RNE^{2}}{N_e * gain^{2}} \]

where $RNE$ is the readnoise per exposure, in electrons, $N_e$ is the number
of exposures per integration, and $gain$ is the detector gain in electrons
per ADU.

\paragraph{UIST Readnoise}

For \UIST\ the formula used depends on the number of reads performed.
For data taken with fewer than 13 reads, the formula used is:

\[   V_{RN} = \frac{( RN * ( -0.0322 * ( N - 1 ) + 1.0322 ) )^{2}}{gain^{2}} \]

For data taken with between 13 and 51 reads the formula used is:

\[   V_{RN} = \frac{( 1.5616 * RN * ( ( N - 1 )^{-0.3568} ) )^{2}}{gain^{2}} \]

For data taken with more than 51 reads the variance due to readnoise
is:

\[   V_{RN} = \frac{225}{gain^{2}} \]

For all formulas $RN$ is the readnoise in electrons, $N$ is the number
of reads, and $gain$ is the detector gain in electrons per ADU.

\paragraph{IRIS2 Readnoise} For \IRIS\ the formula used to determine
the variance due to readnoise is:

\[   V_{RN} = \frac{RNE^{2}}{N_e * gain^{2}} \]

where $RNE$ is the readnoise per exposure, in electrons, $N_e$ is the
number of exposures per integration, and $gain$ is the detector gain
in electrons per ADU.

\paragraph{ISAAC Readnoise} \ISAAC\ uses the same formulae for calculating
the readnoise variance as \Michelle\, {\em i.e.}

\[   RNE = \frac{RN}{\sqrt{\frac{N_{ND} * ( N_{ND} + 1 )}{12 * ( N_{ND} + 1 )}}} \]

and

\[   V_{RN} = \frac{RNE^{2}}{N_e * gain^{2}} \]

where $RNE$ is the readnoise per exposure, in electrons, $RN$ is the
readnoise in electrons, and $N_{ND}$ is the number of non-destructive reads,
$N_e$ is the number
of exposures per integration, and $gain$ is the detector gain in electrons
per ADU.

\subsubsection{\xlabel{bias_subtraction}Bias Subtraction\label{bias_subtraction}}

For observations not taken in non-destructive read mode, a bias frame
is subtracted. The bias frame is pulled from the calibration system,
having been filed using the \htmlref{REDUCE\_BIAS}{REDUCE\_BIAS}
recipe.
\newline {\em[\_SUBTRACT\_BIAS\_]}

\subsubsection{\xlabel{poisson_variance}Poisson Variance\label{poisson_variance}}

Once the readnoise variance has been added and the bias has been optionally
subtracted, the variance due to Poisson noise is added. For all instruments
the Poisson variance is calculated as:

\[   V_{P} = S * gain * N_e \]

where $S$ is the signal in ADU per exposure, $gain$ is the detector
gain in electrons per ADU, and $N_e$ is the number of exposures per
integration.

At this stage the number of pixels that are background limited is
displayed. This number is simply the percentage of pixels where the
Poisson noise is greater than the readnoise.
\newline {\em[\_ADD\_POISSON\_VARIANCE\_]}

\subsubsection{\xlabel{chopping}Chopping\label{chopping}}

In the thermal and mid-infrared regimes the sky is varying so rapidly
that normal reduction methods are inappropriate. Instead sky
subtraction is achieved either by frequently oscillating the secondary
mirror between two beams (mid-infrared), called A and B; or by moving
the telescope offsets (thermal) after a short exposure. The generic
term is {\em chopping}.

Both methods produce frames with the target spectrum on different
rows of the detector. The \htmlref{POINT\_SOURCE}{POINT\_SOURCE}
and \htmlref{EXTENDED\_SOURCE}{EXTENDED\_SOURCE} recipes difference
these pairs of frames so that the result has both a positive and negative
spectrum, and a background close to zero. The sense of the subtraction
is always the same. \ORACDR\ subtracts the B beam
from the A beam, and the normal sequence is ABBA.
\newline{\em[\_SUBTRACT\_CHOP\_]}

\subsubsection{\xlabel{flat_fielding}Flat Fielding\label{flat_fielding}}

Depending on the data format, this step and the subsequent step
(interleaving and coadding, see
\begin{latexonly}
Section~\ref{interleave_and_coadd}
\end{latexonly}
\begin{htmlonly}
\htmlref{Interleaving and Coadding}{interleave_and_coadd}
\end{htmlonly}
) may be swapped. If the flat frame and data frame were both taken
with the same interleaving, then flat-fielding is done after interleaving.
Otherwise, flat-fielding is done first. See
\begin{latexonly}
Section~\ref{flat_fields}
\end{latexonly}
\begin{htmlonly}
\htmlref{Flat Fields}{flat_fields}
\end{htmlonly}
for information on how spectroscopic flat-fields are created.

Flat-fielding is done by a straight division of the data frame by the
appropriate flat-field calibration frame.
\newline {\em[\_DIVIDE\_BY\_FLAT\_, \_FLATFIELD\_COADD\_INTERLEAVE\_]}

\subsubsection{\xlabel{interleave_and_coadd}Interleave and
Coadd\label{interleave_and_coadd}}

In order to fully sample a spectrum and reduce the effect
of bad pixels, observations are often taken at different detector
positions. The detector is stepped along the spectral axis by a
fractional number of pixels; for \CGS this is typically in half-pixel
or third of a pixel steps. Each spectral element can be sampled more
than one time, which helps increase the signal-to-noise and decrease
the impact of hot or bad pixels. This sampling method is often
referred to as 2$x$2 or 3$x$2 sampling. The first number refers to
the number of data points taken per resolution (or the inverse
of the fractional pixel step size) and the second refers to the
number of times each pixel has been observed. These observations need to be
interleaved to create a higher-resolution spectral image.

The interleaving is done by expanding the input frames by the
reciprocal of the fractional pixel step size, then blanking out
the extra columns in these expanded frames. The origins are then
shifted correspondingly, and the expanded frames are coadded
together using the mean to create a final spectral image.
\newline {\em[\_INTERLEAVE\_COADD\_, \_FLATFIELD\_COADD\_INTERLEAVE\_]}

\subsubsection{\xlabel{orient_image}Orient Image Normally\label{orient_image}}

Some instruments are set up such that the spectrum on the detector
runs from higher wavelength to lower wavelength as pixel value in
increased, which is reverse to expectations. This step flips
the image so that the shorter wavelength is to the left and longer
is to the right.
\newline {\em[\_ORIENT\_IMAGE\_NORMALLY\_]}

\subsubsection{\xlabel{wavelength_calibrate}Wavelength Calibrate\label{wavelength_calibrate}}

Wavelength calibration is necessary so that spectral features can
be identified. \ORACDR currently only calculates a wavelength
estimation based on information contained in the FITS headers.
It takes the values for dispersion and central wavelength, then
applies these values as a linear wavelength scale to the spectral
image.

This is often not good enough for accurate wavelength calibration
as most grisms and dispersers produce higher-order dispersions, so
manual wavelength calibration must be performed after spectra have
been extracted.
\newline {\em[\_WAVELENGTH\_CALIBRATE\_BY\_ESTIMATION\_]}

\subsection{\xlabel{group_formation}Group Formation\label{group_formation}}

After the individual frames have been processed, a composite group
spectral image must be formed. Most infrared spectroscopic observations
are taken in object-sky pairs, so the first step in group formation
is subtracting the sky frame from its corresponding object frame.

\subsubsection{\xlabel{sky_subtraction}Sky Subtraction\label{sky_subtraction}}

To be able to do sky subtraction, \ORACDR needs to know out of a
pair of frames which is the object frame and which is the sky frame.
To do this \ORACDR examines the FITS headers. If the offset for both
right ascension and declination are less than 0.001 arcseconds, then
the frame is treated as being on-source. Otherwise, the frame is
off-source and is used as a sky frame.

\IRIS does not record telescope offsets for spectroscopy mode, so
this method cannot be used. Instead \ORACDR examines the aperture
used. If aperture A is used then the frame is on-source and the
right ascension offset is set to zero, otherwise the frame is
off-source and the right ascension offset is set to 26.92 arcseconds.

\ESO instruments are different again as observations are not done
in object-sky pairs. Instead they are done in equal-sized blocks
of object and sky observations, such that a certain number of
object observations are done, followed by an equal number of sky
observations. In this case the initial frame in a group is always
assumed to be on-source. As with the standard pipeline, an observation
is considered to be off-source if its offsets are greater than
0.001 arcseconds.
\newline {\em[\_PAIR\_REDUCTION\_STEER\_]}

Sky subtraction is straightforward -- the sky frame is subtracted
from the object frame.

For \ESO instruments the corresponding sky frame in a block is
subtracted from the respective object frame in a block, such that
the same position in each set is considered as an object-sky pair.
\newline {\em[\_PAIR\_REDUCTION\_SUBTRACT\_]}

\subsubsection{\xlabel{group_coadddition}Group Coaddition\label{group_coaddition}}

Group coaddition is performed by taking the average of all of
the sky-subtracted pairs in the group. The header values for
airmass and UT time at the end of observation are updated in
the group frame.
\newline {\em[\_PAIR\_REDUCTION\_COADD\_TO\_GROUP\_]}

In polarimetry mode multiple group files are created, one for
each waveplate position. Airmass and UT time header values
are also updated.
\newline {\em[\_PAIR\_REDUCTION\_COADD\_TO\_GROUP\_POL\_]}

\subsection{\xlabel{spectrum_extraction}Spectrum Extraction\label{spectrum_extraction}}

\subsubsection{\xlabel{counting_beams}Counting Beams\label{counting_beams}}

The first step in doing spectral extraction is determining the number
of beams to extract. For regular object-sky observations there will
be one positive beam and, depending on how large the offsets are or if
nodding was done along the slit or not, zero or one negative beams. A nod
is considered to be along the slit if the nod angle is within 5 degrees of
the slit angle, which represents roughly 1 arcsecond over a 10 arcsecond
throw. The length of the slit is not taken into account, so throws to
a position off the end of the slit will still count as being along the
slit, even though the spectrum will not appear on the detector.

For chopped observations there can be one or two positive beams and
zero, one or two negative beams, depending on combinations of chop
throw, chop angle, nod throw and nod angle. If the chop throw and nod
throw are equal to within 2 arcseconds and the chop and nod are along the slit,
there will be one positive beam and two negative beams. If the chop
throw and nod throw are equal to within 2 arcseconds and the chop and nod are
to sky, there will be two positive beams and two negative beams. If
the chop is along the slit then there will be one positive beam and
one negative beam. If the chop is along the slit but the offset is
to sky, then there will be one positive beam and one negative beam.
If the chop is to sky and the nod is along the slit, then there
will be one positive beam and one negative beam. If the chop and nod
are both to sky, then there will be one positive beam.

For dual-beam polarimetry observations the number of beams is as above, but
doubled. For single-beam polarimetry the number of beams is as above.
\newline {\em[\_EXTRACT\_DETERMINE\_NBEAMS\_]}

\subsubsection{\xlabel{finding_beams}Finding Beams\label{finding_beams}}

After the number of beams to extract has been determined, it comes time
to locate the beams on the detector. The spectral image first has any
residual bias level removed by subtracting a multiply clipped mean, and
it is then collapsed along the spectral axis to form a profile spectrum.

To find the beams, the profile spectrum is turned into a five-pixel wide
image which is made up of the original profile flanked by symmetric half-
and quarter-strength copies. This step is non-parametric, and can prefer
faint blips over strong beams, although in practice the correct beam
is found.
\newline {\em[\_FIND\_PEAKS\_BY\_MAKING\_IMAGE]}

If the number of beams found does not equal the number of beams calculated
in the previous step (see
\begin{latexonly}
Section~\ref{counting_beams}
\end{latexonly}
\begin{htmlonly}
\htmlref{Counting Beams}{counting_beams}
\end{htmlonly}
) then spectral extraction will not occur. If flux calibration is to
be performed, then processing skips to division by standard
(\ref{division_by_standard}), if division by standard and flux calibration
is necessary.

\IRIS differs in that the entire spectral image is not collapsed to
form the profile. Collapsing the entire image risks producing spurious
peaks due to noisy data near the edges of the array, so a profile is
formed by collapsing a region 0.05 microns short and 0.15 microns long of
the central wavelength.

After the beam locations have been determined they are filed with the
calibration system to be used for faint sources, if necessary.

The beam detection step described here does not modify the Group file.
\newline {\em[\_EXTRACT\_FIND\_ROWS\_]}

\subsubsection{\xlabel{beam_extraction}Beam Extraction\label{beam_extraction}}

Once the beam positions have been located, the beams can be extracted.
First, an extraction window width is calculated based on the position
and number of beams in the spectral image. This window is used for all
beams. If there are two beams, then the half-width of the window is
half the beam separation. If there are three, then the half-width of
the window is half the distance between the positive beam and the first
negative beam. Otherwise, the half-width is 50 pixels.

\UIST differs from this in that if either the HK or KL grisms are
used, the half-width of the extraction window is 10 pixels.

For each beam, an optimal extraction profile (Horne, 1989) is then
determined over the extraction window. If requested (i.e. if a standard
star is reduced), this optimal extraction profile is filed with the
calibration system.

The beams are then optimally extracted using the appropriate profile.
\newline {\em[\_EXTRACT\_ALL\_BEAMS\_]}

\subsubsection{\xlabel{derippling}Derippling\label{derippling}}

When observations are interleaved (see
\begin{latexonly}
Section~\ref{interleave_and_coadd}
\end{latexonly}
\begin{htmlonly}
\htmlref{Interleave and Coadd}{interleave_and_coadd}
\end{htmlonly}
), variations in throughput, sky brightness, and other factors may cause
flux levels to change between the interleaved exposures, resulting in a
ripple in the interleaved spectrum. This ripple is removed by creating a ripple flat with a ripple that has the same frequency as the interleaving, i.e. if 2x2 interleaving is performed, a ripple flat with a frequency of two pixels is created. This ripple flat is divided into the extracted spectrum unless the ripple amplitude is less than 70%.
\newline {\em[\_DERIPPLE\_ALL\_BEAMS\_]}

\subsubsection{\xlabel{beam_cross_correlation}Beam Cross Correlation\label{beam_cross_correlation}}

To remove any shift in beams, possibly caused by tilted spectra, the
extracted beams are cross-correlated and shifted. If the maximum
value of the cross correlation funtion is less than 0.6, or if the
shift is greater than 2 pixels, then the spectra are not aligned
and shifted.
\newline {\em[\_CROSS\_CORR\_ALL\_BEAMS\_]}

\subsubsection{\xlabel{extracted_beam_coaddition}Extracted Beam Coaddition\label{extracted_beam_coaddition}}

The beams have been extracted and must now be coadded. This is done
by simply averaging all of the extracted beams together.
\newline {\em[\_COADD\_EXTRACTED\_BEAMS\_]}

\subsection{\xlabel{snr_calculation}Signal-To-Noise Calculation\label{snr_calculation}}

When an extracted and co-added spectrum has been determined, the
signal-to-noise ratio is calculated, but only if it has a variance
array associated with it.
\newline {\em[\_CALCULATE\_SNR\_]}

\subsection{\xlabel{division_by_standard}Division By Standard Star\label{division_by_standard}}

Whether or not a spectrum has been extracted, division by a standard star
spectrum can still proceed. If no spectrum has been extracted, then the
standard star spectrum is extended perpendicularly to the wavelength direction
to make a 2-D spectrum.

When the spectrum of the science target is divided by a standard star spectrum, a straight division is done.

\subsection{\xlabel{flux_calibration}Flux Calibration\label{flux_calibration}}

After standard division is done, the spectrum is flux-calibrated. This is
done by using the standard star's V-band magnitude, spectral type, and
observed waveband to determine a reference flux at the central wavelength
of the observed waveband. The units of the flux-calibrated spectrum are
presented in $W/m^{2}/{\mu}m$.

\section{\xlabel{science_target_recipe_details}Science Target Recipe Details\label{science_target_recipe_details}}

There are three basic recipes available for reducing science target observations: EXTENDED\_SOURCE, POINT\_SOURCE, and FAINT\_POINT\_SOURCE. These three have many variants that dictate the use or lack of calibration data. For example, there exists a POINT\_SOURCE\_NOFLAT recipe that is meant to be used for point source reduction without flat-fielding. The use of these variants is typically avoided, as the basic recipes can often flag uncalibratable data.

In addition, dual-beam polarimetry can be reduced using the POINT\_SOURCE\_POL recipe. This recipe is a variant of the POINT\_SOURCE recipe.

\subsection{\xlabel{point_source}POINT\_SOURCE\label{point_source}}

The POINT\_SOURCE recipe is used to reduce observations of point sources. It reduces data using all of the steps outlined in
\begin{latexonly}
Section~\ref{features_of_the_primitives}
\end{latexonly}
\begin{htmlonly}
\htmlref{Features of the Primitives}{features_of_the_primitives}
\end{htmlonly}
.

The following variants are available: POINT\_SOURCE\_NOARC (no arc spectrum extracted), POINT\_SOURCE\_NOFLAT (no division by a flat-field), POINT\_SOURCE\_NOFLAT\_NOARC, POINT\_SOURCE\_NOFLAT\_NOSTD, and POINT\_SOURCE\_NOSTD (no division by a standard star).

\subsection{\xlabel{extended_source}EXTENDED\_SOURCE\label{extended_source}}

The EXTENDED\_SOURCE recipe is used to reduce observations of extended sources. It differs from
\begin{htmlonly}
\htmlref{POINT\_SOURCE}{point_source}
\end{htmlonly}
in that EXTENDED\_SOURCE skips
\begin{htmlonly}
\htmlref{spectrum extraction}{spectrum_extraction}
\end{htmlonly}
and proceeds directly with
\begin{htmlonly}
\htmlref{signal-to-noise calculation}{signal_to_noise_calculation}
\end{htmlonly}
and
\begin{htmlonly}
\htmlref{division by a standard star}{division_by_standard_star}
\end{htmlonly}
.

The following variants are available: EXTENDED\_SOURCE\_NOARC (no arc spectrum extracted), EXTENDED\_SOURCE\_NOFLAT (no division by a flat-field), EXTENDED\_SOURCE\_NOFLAT\_NOARC, EXTENDED\_SOURCE\_NOFLAT\_NOSTD, and EXTENDED\_SOURCE\_NOSTD.

\subsubsection{\xlabel{extended_source_with_separate_sky}EXTENDED\_SOURCE\_WITH\_SEPARATE\_SKY\label{extended_source_with_separate_sky}}

In addition to the variants to EXTENDED\_SOURCE listed above, EXTENDED\_SOURCE\_WITH\_SEPARATE\_SKY exists for very extended sources where it is not possible to nod along the slit, thus a separate sky is used for sky-subtraction. The separate sky is reduced using the
\begin{htmlonly}
\htmlref{REDUCE\_SKY}{reduce_sky}
\end{htmlonly}
calibration recipe.

The following variants are available: EXTENDED\_SOURCE\_WITH\_SEPARATE\_SKY\_NOFLAT (no division by a flat-field), EXTENDED\_SOURCE\_WITH\_SEPARATE\_SKY\_NOFLAT\_NOSTD (no division by a flat-field or by a standard star), and EXTENDED\_SOURCE\_WITH\_SEPARATE\_SKY\_NOSTD (no division by a standard star).

\subsection{\xlabel{faint_point_source}FAINT\_POINT\_SOURCE\label{faint_point_source}}

The FAINT\_POINT\_SOURCE recipe is used to reduce observations of point sources that are faint enough that they may not be detected in the
\begin{htmlonly}
\htmlref{beam detection.}{beam_detection}
\end{htmlonly}
\begin{latexonly}
beam detection step described in Section~\ref{beam_detection}.
\end{latexonly}
In this case, the locations of the beams as determined for the standard star are used to extract the object. Optimal extraction is still used.

The following variants are available: FAINT\_POINT\_SOURCE\_NOFLAT (no division by a flat-field), FAINT\_POINT\_SOURCE\_NOFLAT\_NOSTD (no fivision by a flat-field and no extraction of an arc spectrum), and FAINT\_POINT\_SOURCE\_NOSTD (no extraction of an arc spectrum).

\subsection{\xlabel{point_source_pol}POINT\_SOURCE\_POL\label{point_source_pol}}

Dual-beam spectropolarimetry data can be reduced using the POINT\_SOURCE\_POL recipe. This recipe reduces data using the steps in Sections~\ref{preparation_of_single_frames}, \ref{group_formation}, and \ref{spectrum_extraction}. The data can be taken in any order between waveplate positions, but corresponding object-sky pairs for a given waveplate position must be observed together. Spectrum extraction is done such that separate $E$ and $O$ beam spectra are created for each waveplate position. After a cycle of eight observations has done, corresponding to one object-sky pair at each of the 0-, 45-, 22.5-, and 67.5-degree waveplate angles, the pipeline calculates the $I$, $Q$, and $U$ Stokes parameters, from which the percentage polarization, polarization intensity, and polarization angle spectra are calculated.

\section{\xlabel{calibration_recipe_details}Calibration Recipe Details\label{calibration_recipe_details}}



\end{document}
